Sources/SwiftGRPCNIOInteroperabilityTests/InteroperabilityTestCases.swift

```diff
@@ -0,0 +1,679 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+import SwiftGRPCNIO
+import NIOHTTP1
+
+/// This test verifies that implementations support zero-size messages. Ideally, client
+/// implementations would verify that the request and response were zero bytes serialized, but
+/// this is generally prohibitive to perform, so is not required.
+///
+/// Server features:
+/// - EmptyCall
+///
+/// Procedure:
+/// 1. Client calls EmptyCall with the default Empty message
+///
+/// Client asserts:
+/// - call was successful
+/// - response is non-null
+///
+/// It may be possible to use UnaryCall instead of EmptyCall, but it is harder to ensure that the
+/// proto serialized to zero bytes.
+class EmptyUnary: InteroperabilityTest {
+  func run(using connection: GRPCClientConnection) throws {
+    let client = Grpc_Testing_TestServiceService_NIOClient(connection: connection)
+    let call = client.emptyCall(Grpc_Testing_Empty())
+
+    try assertEqual(call.response, Grpc_Testing_Empty())
```

Comment:
Seeing this without a `.wait()` after `call.response` confused me at first. Personally, I'd prefer to make the `.wait()` explicit here (and elsewhere, of course), but not feeling strongly given that this is "only" test code.

Thinking about this more, I agree that cluttering the tests with `try ....wait()` statements would probably be annoying.

Reply:
That's the same thought process I went through! I think some renaming could help:

```swift
try assertEqual(future: call.response, Grpc_Testing_Empty())
// or...
try assertEqual(future: call.response, value: Grpc_Testing_Empty())
```

For two futures:
```swift
try assertEqual(future: call1.response, future: call2.response)
// or maybe
try assertFuturesEqual(call1.response, call2.response)
```

It might be worth explicitly calling these out as synchronous as well (`assertEqualSync` or similar).

What do you think?